### PR TITLE
10.0 pos debt issues

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
Unable to purchase a credit
1. Add 'Credit top-up' for $100
2. Choose a customer
3. Select Cash payment method for $100
Result: a message 'Use an another non-discount payment method'

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
Cannot withdraw from Credit (Friuts&Vegatables) using "Pay Debt" product
1. Edit the payment method : disable "Zero transaction", enable "Allow to cash out credits" .
2. Add the product "Pay Debt", select a customer, click "Payment"
3. Select the payment method , set a value
4. Select a payment method where to withdraw. Click "Validate".
Result: error message "You may only buy Fruits and Vegetables..."

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
Can withdraw from a credit more than available
1. Edit the payment method : disable "Zero transaction", disable "Allow to cash out credits" .
2. Add the product "Pay Debt", select a customer, click "Payment"
3. Select the payment method , set a value more than available
4. Select a payment method where to withdraw. Click "Validate".
    Result: Success. Should not to be.
I transfered from "Credit" to "Credit (Fruits & Vegetables)".

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
1. Open POS
2. Select a product that is not from the category "Fruits & Vegatable" (Acsone.eu 1 Unit)
3. Select a customer
4. Click Payment
5. Click on Credit (Friuts & Vegetables only)
6. Payment
Result: succeess. POS should cause an error.

File: pos_debt_notebook/static/src/js/pos.js [hunk 1]
```diff
@@ -142,22 +141,23 @@ odoo.define('pos_debt_notebook.pos', function (require) {
...
-                            self.reload_debts(load_partner_ids, 0, {"postpone": 4000, "shadow": false});
+                            self.reload_debts(load_partner_ids, 0, {"postpone": 4000, "shadow": false, 'deferred': download_debts_ready});
                             self.trigger('updateDebtHistoryFail');
                         });
                         return request_finished;
                     });
                 }
             });
-            return this.reload_debts_ready;
+            return download_debts_ready;
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js:160]: This update is not covered in changelog

File: pos_debt_notebook/static/src/js/pos.js [hunk 2]
```diff
@@ -533,11 +559,33 @@ odoo.define('pos_debt_notebook.pos', function (require) {
...
+            var allowed_products = _.filter(order.get_orderlines(), function(ol){
+                var categories = ol.product.pos_categ_id
+                ? [ol.product.pos_categ_id[0]]
+                : ol.product.pos_category_ids;
+                if (_.intersection(cr.journal.category_ids, categories).length) {
+                    return true;
+                }
+                return false;
+            });
+            if (allowed_products.length === order.get_orderlines().length || this.cash_out_check()) {
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: It's actually allowed_lines rather than allowed products

**Review by yelizariev — APPROVED** [review]

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

**Pull request merged**